OneCommunityGlobal/HighestGoodNetworkApp
# Aditya Sure - Resolved Bell Notification for Hours Due in 48 Hours

# Description:

This pull request resolves the bell notification issue for hours due in 48 hours.

![image](https://github.com/user-attachments/assets/c81d6a55-01ac-42d4-af5b-81f524f062b8)

## Related PRS (if any):

None

## Main changes explained:
- Created a NotificationBell component to handle bell notification logic.
- Integrated the notification logic within the existing system to ensure it triggers correctly based on the user’s time logged and time remaining.

## How to test:

1. Copy the Code: Apologies, but I’m currently unable to push changes directly to the main repository due to permission issues. Please copy the code from the files changed in this pull request.
2. Run npm install and any other necessary commands to prepare the environment.
3. Clear site data/cache to ensure no previous notifications interfere.
4. Log in as any user.
5. Navigate to the Dashboard and check the Bell to verify the notification functionality.
6. Use the "Add Intangible Hours" function in the Timelog to test the feature by manipulating the time.
7. Verify that the bell notification triggers appropriately when less than 50% of the hours are completed and only 48 hours remain.
8. Verify that the notification resets after it is viewed or when the week ends.
9. Confirm that the feature works correctly in both light and dark mode.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/64a1d65e-1c16-4459-9a0c-9c993bdf2505

Add Intangible Time Entry Button See Below:

![image](https://github.com/user-attachments/assets/f6c7b8d9-1919-466b-99f0-826f3a27a4a7)

Press the Button and add/edit/delete Time:

![image](https://github.com/user-attachments/assets/7361a11e-80c1-4196-8d7f-0c3f72e1fed3)

## Note:
1. Modify Time Left:
Locate the following code snippet in the NotificationBell component:
![image](https://github.com/user-attachments/assets/89be01d2-551a-4740-98ad-3429205cd784)
Change the 48 in hoursLeft <= 48 to any other number (e.g., 100) to test different time thresholds for when the notification should trigger.
2. Simulate Time Log via Timelog:
Use the "Add Intangible Hours" feature in the Timelog section to simulate logged hours.
Add, edit, or delete hours to manipulate and test the notification behavior.
3. Clear Cookies and LocalStorage for Retesting:
The notification only triggers once per week, unless conditions are met again. Clear your browser's cookies or localStorage to reset the notification for retesting.
This ensures that previous notifications don't interfere with the current test.

# Additional Note:
Apologies for the inconvenience. I’m currently unable to push changes directly to the main repository due to permission issues. I’ve contacted Jae Sabol regarding this, and once I receive the necessary permissions to create a new branch, I will update you. In the meantime, please copy the code from the files changed in this pull request to test the updates. If you need any assistance or have questions, feel free to reach out. I’m here to help!
